clusterDeployer.py

```diff
@@ -119,8 +119,12 @@ def teardown(self) -> None:
         removed_macs = []
         names = [x.name for x in self._cc.all_vms()]
         ips = [x.ip for x in self._cc.all_vms()]
-        for e in q[-1][0][1:]:
-            if e.attrib["name"] in names or e.attrib["ip"] in ips:
+        dhcp = None
+        ip_tree = q.find('ip')
+        if ip_tree:
+            dhcp = ip_tree.find('dhcp')
+        for e in dhcp or []:
```

Comment:
Why not init `dhcp` to `[]` ?

Reply:
ip_tree.find('dhcp')  returns None if 'dhcp' is not found in in_tree.
So initializing dhcp to Null means "'dhcp' is not found".